lib/reverse_proxy/client.rb

```diff
@@ -86,7 +86,7 @@ def request(env, options = {}, &block)
 
       # Make the request
       Net::HTTP.start(uri.hostname, uri.port, http_options) do |http|
-        callbacks[:on_connect].call(http)
+        callbacks[:on_connect].call(http, target_request)
```

Comment:
Nice, I wonder if we should update the readme with a comment about how this fork differs from the base. I don't know what the best practices are when making private (or public in this case) forks of a main repo.

Reply:
Its a good idea to add something to the ReadMe since this is changing the function signature and is a breaking change. However there might be more changes needed so I have not summarized the changes yet.